File: Data Structures & Algorithms/counting-bits/submission-0.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> res(n + 1);
        for (int i = 0; i <= n; i++)
        {
            int num = i;
            int count = 0;
            while (num)
            {
                num &= (num - 1);
                count++;
            }
            res[i] = count;
        }
        return res;
    }
};

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        res = []

        def dfs(i, currlist):
            if len(currlist) == len(nums):
                res.append(currlist.copy())
                return

            for i, num in enumerate(nums):
                if num in currlist:
                    continue

                currlist.append(num)
                dfs(i, currlist)
                currlist.pop()            

        dfs(0, [])
        return res